### PR TITLE
Use cancellation token and exit notifications instead of watch

### PR DESCRIPTION
**Pull request by ParthDesai**
## Description
This PR changes `call_collate_on` to use cancellation token and exit notification to stop it gracefully as well as wait for it to exit.

### CODE REVIEW

**Review by girazoki — APPROVED** [review]
It looks to me but I would rather have @tmpolaczyk take a look before merging

File: client/consensus/src/collators/lookahead.rs [hunk 1]
```diff
@@ -178,262 +184,256 @@ where
...
-                Ok(None) => continue,
-                Ok(Some(pvd)) => pvd.max_pov_size,
-                Err(err) => {
-                    tracing::error!(target: crate::LOG_TARGET, ?err, "Failed to gather information from relay-client");
-                    continue;
-                }
-            };
+        loop {
+            select! {
+                Some(relay_parent_header) = import_notifications.next() => {
```
Comment (tmpolaczyk) [inline client/consensus/src/collators/lookahead.rs:189]: The previous code exited the loop if `import_notifications.next()` returned `None`, what will happen now?

**Review by tmpolaczyk — COMMENTED** [review]
Some comments but overall it looks better than what we had before

File: client/consensus/src/collators/lookahead.rs [hunk 2]
```diff
@@ -142,7 +146,9 @@ where
 
     log::info!("LOOKAHEAD COLLATOR RUNNING...");
```
Comment (tmpolaczyk) [inline client/consensus/src/collators/lookahead.rs:147]:
```suggestion
```

This was probably meant for debugging

**Review by tmpolaczyk — APPROVED** [review]

File: node/src/container_chain_spawner.rs [hunk 3]
```diff
@@ -135,8 +139,21 @@ impl ContainerChainSpawner {
...
+            self.chain_type.clone(),
+            self.relay_chain.clone(),
+            self.relay_chain_interface.clone(),
+            self.collator_key.clone(),
+            self.sync_keystore.clone(),
+            self.orchestrator_para_id,
+            self.validator,
+            self.spawn_handle.clone(),
+            self.state.clone(),
+        );
```
[the remark this answers is not in the document]
Comment (ParthDesai) [reply node/src/container_chain_spawner.rs:156]: I tried to find more idiomatic way than that and at last left it as it is. Will open an issue and improve this as part of next PR.

**Pull request merged**